cpp/ql/src/Best Practices/Unused Entities/UnusedStaticFunctions.ql

```diff
@@ -13,16 +13,34 @@
 
 import cpp
 
+pragma[noinline]
+predicate possiblyIncompleteFile(File f) {
+  exists(Diagnostic d | d.getFile() = f and d.getSeverity() >= 3)
+  or
+  exists(ErrorExpr ee | ee.getFile() = f)
```

Comment:
This is a very bad indicator of an incomplete file for the same reason an exception is made in the IR translation here: https://github.com/github/codeql/blob/main/cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedElement.qll#L116-L119

Reply:
Oh, good point! I had totally forgotten about that special case. Thanks for the heads up. Now go back to your vacation 😂!

Reply:
I think this settles that we only want to keep the `Diagnostic`s case.  My testing supports that this was generally the better indicator in practice anyway.

Though I will note there are several other queries that take the presence of an `ErrorExpr` to indicate incomplete extraction.